### PR TITLE
Stop overriding completion_time when finishing span

### PR DESCRIPTION
**Pull request by christopheonce**
This should solve #55 and allow creating span based on logs.

### CODE REVIEW

**Comment by zachdaniel** [comment]
I also have a branch up with a fix :) I'm happy to go with yours though. Make sure to account for the change in `finish_trace` also.

File: lib/spandex.ex [hunk 1]
```diff
@@ -157,13 +157,19 @@ defmodule Spandex do
...
-        finished_span = update_or_keep(span, completion_time: adapter.now())
+        finished_span = ensure_completion_time_set(span, adapter)
 
         strategy.put_trace(opts[:tracer], %{trace | stack: tail, spans: [finished_span | spans]})
         {:ok, finished_span}
     end
   end
 
+  defp ensure_completion_time_set(span = %Span{completion_time: nil}, adapter) do
+    update_or_keep(span, completion_time: adapter.now()) |> ensure_completion_time_set(adapter)
```
[the remark this answers is not in the document]
Comment (zachdaniel) [reply lib/spandex.ex]: This probably doesn't need to be recursive, does it?

**Comment by zachdaniel** [comment]
Everything looks good to me, except that calling `ensure_completion_time/2` from the first clause of `ensure_completion_time/2` seems unnecessary.

**Comment by zachdaniel** [comment]
@christopheonce Ready to merge?

**Comment by christopheonce** [comment]
@zachdaniel I am done with this Pull Request

**Comment by zachdaniel** [comment]
@christopheonce Thanks for your contribution! This is now available (along with the recent adapter move) as version 2.0.0 in hex.

**Pull request merged**